### DISCUSSION TITLE
Update existing rows with default value

### DISCUSSION THREAD

**Discussion by standuprey** [mikro-orm/mikro-orm#2471] [Ideas]
**Is your feature request related to a problem? Please describe.**
If I create a new column like this
```
  @Property()
  status: string = 'new';
```
And I have existing rows for this table, the migration will be:
```
this.addSql('alter table "store" add column "status" varchar(255) not null;');
```
And it will fail

**Describe the solution you'd like**
A migration that would work is
```
'alter table "shopify" add column "status" varchar(255) not null default(\'new\');'
```
Also you might not want to update existing row with the default value, so an optional `existingDefault` could be valuable
In my example, I could have `@Property({existingDefault: 'active'})` generating
```
this.addSql('alter table "store" add column "status" varchar(255) null default(\'new\');');
this.addSql('update "store" set "status" = \'active\'');
this.addSql('alter table "store" alter column "status" set not null;');
```

**Describe alternatives you've considered**
I have been doing this manually, but thought that could be a nice feature

(Note that I am working with pg)

**Comment by B4nan** [comment]
To get the value from initializer, we would need to create entity instance via ctor, and that might not be always possible due to some dependencies in the constructor. It might even have sideeffects to call the entity ctor (although it would be very weird ctor implementation, but it is possible). With ts-morph this would be obviously possible just via reflection.

**Comment by rodrigovilar** [comment]
I suggest to use the `@Property({existingDefault: 'active'})` form.
You will not need to create an instance to get this metadata and you can act in the constructor and in the migration in order to implement the default value feature, for memory and DB respectively.

**Comment by B4nan** [comment]
To me that does not sounds much better than explicit `default` option in the decorator (definitely less readable).

Also the value might not be even usable (e.g. `new Date()` or `uuid.v4()`).

**Comment by standuprey** [comment]
`default` is for new rows, `existingDefault` is for exiting rows (ie. rows that were in the DB before the migration)

**Comment by B4nan** [comment]
I see, but that means both of this requires to create new entity instance to extract the value, so not a general solution, rather a hack...

> You will not need to create an instance to get this metadata and you can act in the constructor and in the migration in order to implement the default value feature, for memory and DB respectively.

I don't really understand how you expect to get the actual default values - it is still needed to create the entity instance, as the values are effectively just assignemnts done inside the entity ctor. There is no other way to get those (only via ts-morph reflection, which has its own group of problems). 

Also as a rule of thumb it is not suggested to mix DDL with DML queries (`alter table` vs `update table`).

On the other hand I can confirm I do this myself sometimes, especially when I want to seed some data via `nativeInsert`, the db defaults are needed. It just feels to quirky, I can already see the issues of people asking why it is not working for them. For me there is much bigger value in allowing the use of ctors that in this.
